libraries/Microsoft.Bot.Connector/Authentication/BotFrameworkAuthentication.cs

```diff
@@ -14,6 +14,28 @@ namespace Microsoft.Bot.Connector.Authentication
     /// </summary>
     public abstract class BotFrameworkAuthentication
     {
+        /// <summary>
+        /// Check whether the channel is Government.
+        /// </summary>
+        /// <returns>true if Government; false otherwise.</returns>
+        public abstract bool IsGovernment();
```

Comment:
John worked hard to get away from this type of code and one of the key concepts behind the BotFrameworkAuthenication design is that abstractions such as "IsGovernment" don't exist. With the implementations laid out in the PR, the number of BFAuthentication subclasses expand with the # of sovereign "gov" clouds supported by Azure, which is suboptimal. Can you explain why this change is necessary?

When a BFAuthenication instance is created, it should have been hydrated with all the necessary information it needs for communicating with a recipient. When the instance is used it already has the information it needs, so ideally the calling code doesn't need to know which cloud it's in.

Reply:
I had a meeting with John to get some more context and I have removed the Government specific methods that I exposed.

---
In reply to: [595429131](https://github.com/microsoft/botbuilder-dotnet/pull/5304#discussion_r595429131) [](ancestors = 595429131)